src/user.rs

```diff
@@ -158,33 +159,27 @@ impl<'a> SingleUserRequestBuilder<'a> {
     /// Retrieve all tracks uploaded by the user
     ///
     /// Returns:
-    ///     a list of tracks or an error if one occurred.
-    pub async fn tracks(&mut self) -> Result<Option<Vec<Track>>> {
-        let path = &format!("/users/{}/tracks", self.id.to_string());
-        self.client.get_collections(&path).await
+    ///     an instance of Tracks
+    pub fn tracks(&self) -> Tracks {
+        Tracks::new(self.client.clone(), self.id)
     }
 
     /// Retrieve all tracks liked by the user
     ///
-    /// Note:
-    ///     /users/{id}/favorites endpoint does not expose playlists liked by the user.
-    ///
     /// Returns:
-    ///     a list of tracks if there are any results, None otherwise,
-    ///     or an error if one occurred.
-    pub async fn likes(&mut self) -> Result<Option<Vec<Track>>> {
-        let path = format!("/users/{}/favorites", self.id.to_string());
-        self.client.get_collections(&path).await
+    ///     an instance of Likes
+    pub fn likes(&mut self) -> Likes {
+        Likes::new(self.client.clone(), self.id)
     }
 
-    /// Retrieve all playlists uploaded by the user
-    ///
-    /// Returns:
-    ///     a list of playlists or an error if one occurred.
-    pub async fn playlists(&mut self) -> Result<Option<Vec<Playlist>>> {
-        let path = format!("/users/{}/playlists", self.id.to_string());
-        self.client.get_collections(&path).await
-    }
+    ///// Retrieve all playlists uploaded by the user
+    /////
+    ///// Returns:
+    /////     a list of playlists or an error if one occurred.
+    //pub async fn playlists(&mut self) -> Result<Vec<Playlist>> {
```

Comment:
Are you going to implement the playlists method with the new stream api?

Reply:
yes, I wanted your approval before implementing